tests/covalent_tests/executor/base_test.py

```diff
@@ -331,3 +331,41 @@ async def fake_run(function, args, kwargs, task_metadata):
     metadata, stdout, stderr = asyncio.run(awaitable)
     task_metadata = {"dispatch_id": dispatch_id, "node_id": node_id, "results_dir": results_dir}
     assert metadata == task_metadata
+
+
+def test_async_write_streams_to_file(mocker):
+    """Test write log streams to file method in BaseAsyncExecutor via LocalExecutor."""
+
+    import asyncio
+
+    me = MockAsyncExecutor()
+
+    # Case 1 - Check that relative log files that are written to are constructed in the results directory that is explicitly passed as an argument.
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_file = "relative.log"
+        write_awaitable = me.write_streams_to_file(
+            stream_strings=["relative"], filepaths=[tmp_file], dispatch_id="", results_dir=tmp_dir
+        )
+        asyncio.run(write_awaitable)
```

Comment:
Instead of `asyncio.run` you can probably mark this whole test as a `@pytest.mark.asyncio` function, make it `async def` and then `await write_awaitable` here. If that doesn't work, then this is fine too.